File: service/src/main/java/com/boxfox/cross/service/coin/CoinService.java
package com.boxfox.cross.service.coin;

import com.boxfox.cross.service.JooqReactiveService;
import com.linkbit.android.entity.CoinModel;
import io.reactivex.Single;
import java.util.List;

public class CoinService extends JooqReactiveService implements CoinUsecase {

  private CoinServiceImpl impl;

  public CoinService() {
    this.impl = new CoinServiceImpl();
  }

  @Override
  public Single<List<CoinModel>> getAllCoins() {
    return createSingle(ctx -> impl.getAllCoins(ctx));
  }
}

File: service/src/main/java/com/boxfox/cross/service/coin/CoinService.kt
package com.boxfox.cross.service.coin

import com.boxfox.cross.common.entity.CoinModel
import com.boxfox.cross.service.JooqReactiveService
import io.reactivex.Single

class CoinService(private val impl: CoinServiceImpl = CoinServiceImpl()) : JooqReactiveService() {

    val list: Single<List<CoinModel>> = single { ctx -> impl.getAllCoins(ctx) }
}
